src/ui/index.js

```diff
@@ -77,12 +87,74 @@ export default async (kbnServer, server, config) => {
       },
       vars: defaults(app.getInjectedVars() || {}, uiExports.defaultInjectedVars),
     };
-
+    if (kibanaTranslations.length <= 0) {
+      const language = await getTranslationLanguage(acceptLanguages);
+      kibanaTranslations = await i18nPlugin.getRegisteredLanguageTranslations(language);
+    }
     return this.view(app.templateName, {
       app: app,
       loadingGif: loadingGif,
       kibanaPayload: payload,
       bundlePath: `${config.get('server.basePath')}/bundles`,
+      welcomeMessage: translate('CORE-WELCOME_MESSAGE'),
+      welcomeError: translate('CORE-WELCOME_ERROR'),
     });
   });
 };
+
+function translate(key) {
+  if (!(key in kibanaTranslations)) {
+    return null;
+  }
+  return kibanaTranslations[key];
+}
+
+async function getTranslationLanguage(acceptLanguages) {
+  let langStr = '';
+  let foundLang = false;
+
+  if (acceptLanguages === null || acceptLanguages.length <= 0) {
+    return DEFAULT_LANGUAGE;
+  }
+
+  const acceptLangsLen = acceptLanguages.length;
+  const registeredLanguages = await i18nPlugin.getRegisteredTranslationLanguages();
+
+  for (let indx = 0; indx < acceptLangsLen; indx++) {
+    const language = acceptLanguages[indx];
+    if (language.region) {
+      langStr = language.code + '-' + language.region;
+    } else {
+      langStr = language.code;
+    }
+    if (registeredLanguages.indexOf(langStr) > -1) {
+      foundLang = true;
+      break;
+    }
+  }
+  if (foundLang) {
+    return langStr;
+  }
+
+  const regLangsLen = registeredLanguages.length;
+  for (let indx = 0; indx < acceptLangsLen; indx++) {
+    const language = acceptLanguages[indx];
+    langStr = language.code;
+    for (let regIndx = 0; regIndx < regLangsLen; regIndx++) {
+      const lang = registeredLanguages[regIndx];
+      if (lang.match('^' + langStr)) {
```

Comment:
hm - not good here actually - because for example `ro` is the code for the Romanian language and `rof` for the Rombo language. This negotiation is probably best in an external module such as `negotiator`